### PR TITLE
Update ruby to 3.3-rc

### PR DESCRIPTION
**Pull request by juniortaeza**
This issue is now complete.

https://github.com/cyberark/conjur-puppet/pull/259

### CODE REVIEW

File: ci/Dockerfile.pdk [hunk 1]
```diff
@@ -1,4 +1,4 @@
-FROM ruby:3.2.2
+FROM ruby:3.3-rc
```
Comment (rpothier) [inline ci/Dockerfile.pdk:1]: What is the rc version?

Comment (juniortaeza) [reply ci/Dockerfile.pdk:1]: ruby:3.3-rc is one of the most up-to-date images recently released ––

after re-reviewing the most recent scans in Snyk, our current ruby:3.2.2 reveals 132 (all low) vulnerabilities. ruby:3.3-rc drastically reduces this to 31 (all low) vulnerabilities

**Comment by juniortaeza** [comment]
Thank you for the follow-up draft PR @gl-johnson! however it looks like the build is still processing, and has been for the past 5 days..

**Comment by gl-johnson** [comment]
> Thank you for the follow-up draft PR @gl-johnson! however it looks like the build is still processing, and has been for the past 5 days..

There's an issue with provisioning Windows nodes in Jenkins. I believe infra is still investigating, so this may be blocked for now along with my PR

**Comment by juniortaeza** [comment]
Thank you for letting me know, I'll be on standby in the meantime for this particular project until the Windows nodes are resolved

**Pull request closed without merging**